packages/features/bookings/Booker/components/AvailableTimeSlots.tsx

```diff
@@ -34,30 +44,20 @@ export const AvailableTimeSlots = ({ extraDays, limitHeight, seatsPerTimeslot }:
   };
 
   const schedule = useScheduleForEvent({
-    prefetchNextMonth: !!extraDays && dayjs(date).month() !== dayjs(date).add(extraDays, "day").month(),
+    prefetchNextMonth,
+    monthCount,
   });
+  const nonEmptyScheduleDays = useNonEmptyScheduleDays(schedule?.data?.slots);
 
   // Creates an array of dates to fetch slots for.
   // If `extraDays` is passed in, we will extend the array with the next `extraDays` days.
-  const dates = useMemo(
-    () =>
-      !extraDays
+  const dates = !extraDays
```

Comment:
https://github.com/calcom/cal.com/assets/123581387/1f0cedbb-4b4c-4697-868c-df9008167fe4

Because of using useMemo sometimes the data isn't fetching properly. since `[date, extraDays, nonEmptyScheduleDays]` are same. After removal it's working as expected.


https://github.com/calcom/cal.com/assets/123581387/4d8b3f85-7abb-41c9-8404-9554b5ca5272

Note : make sure u have 1 slot available per week.